app/policies/division_policy.rb

```diff
@@ -18,4 +18,30 @@ def update?
   def destroy?
     division_admin(division: @record)
   end
+
+  def select?
+    show?
+  end
+
+  class Scope < Scope
+    def resolve
+      scope.where(id: accessible_ids)
+    end
+
+    # This merges in child divisions of the divisions for which a user has been specifically
+    # granted access.
+    def accessible_ids
+      all_ids = base_accessible_ids.map do |id|
+        division = Division.find_safe(id)
+        division.self_and_descendants.pluck(:id) if division
+      end
+      all_ids.flatten.uniq.compact
```

Comment:
No

sent via wireless telephone machine.
On Apr 29, 2016 01:27, "Joseph Elliott" <notifications@github.com> wrote:

> In app/policies/division_policy.rb
> <https://github.com/sassafrastech/madeline_system/pull/68#discussion_r61535576>
> :
>
> > +    show?
> > +  end
> > +
> > +  class Scope < Scope
> > +    def resolve
> > +      scope.where(id: accessible_ids)
> > +    end
> > +
> > +    # This merges in child divisions of the divisions for which a user has been specifically
> > +    # granted access.
> > +    def accessible_ids
> > +      all_ids = base_accessible_ids.map do |id|
> > +        division = Division.find_safe(id)
> > +        division.self_and_descendants.pluck(:id) if division
> > +      end
> > +      all_ids.flatten.uniq.compact
>
> Does it make a difference if the compact is before or after the uniq?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/sassafrastech/madeline_system/pull/68/files/68b8e720304d3a40764283d8e9555d508a11c1ca#r61535576>
>